public/js/submit.js

```diff
@@ -1,132 +1,44 @@
 var successMessage = document.getElementById("submitSuccessMessage");
 var errorMessage = document.getElementById("submitErrorMessage");
 
 var floatingForms = document.getElementsByClassName("form-floating");
 var postForms = document.getElementsByClassName("post-form");
 
-var form = document.forms["form"];
-
 var post = [];
 
-successMessage.style.display = 'none';
-errorMessage.style.display = 'none';
+successMessage.style.display = "none";
+errorMessage.style.display = "none";
+
 
 // specific element will be verified
 //the caller page give the specific var anfd specific function thing
-
-sendDataOnClick(post);
-
-function validateForm()
+if(typeof(url) !== "undefined")
 {
-  $validateForm = true;
-  for(postForm of postForms)
-  {
-    var editText = (postForm.getElementsByClassName("form-control"))[0];
-    var invalid = (postForm.getElementsByClassName("invalid-feedback"))[0];
-    if(! validateType(editText))
-    {
-      console.log("erreur validateType");
-      $validateForm = false;
-      invalid.style.display = 'block';
-    }
-    else
-    {
-      post[ editText['name']] =   editText['value'] ;
-      console.log(post);
-    }
-  };
-  return $validateForm;
+  sendDataOnClick(post, url);
 }
 
-function validateType(editText)
-{
-  $validateType = true;
-  $value = editText['value'];
-  $type = editText['type'];
-  
-  //textare part
-  if(editText.tagName === 'TEXTAREA')
-  {
-    if($value.length < 3)
-      {
-        $validateType = false;
-      }
-  }
-  
-  //input part
-  if(editText.tagName === 'INPUT')
-  {
-    switch($type)
-    {
-      case 'text':
-        if($value === '')
-        {
-          $validateType = false;
-        }
-        break;  
-        case 'password':
-          if($value.length < 6)
-          {
-            $validateType = false;
-          }
-          break;
-          case 'email':
-            $validateType = false;
-            if($value.length >= 6 )
-            {
-              $values = $value.split("@");
-              if($values.length === 2)
-              {
-                //start something1@something2.something3
-                //after the split
-          // $values is like something1,something2.something3 
-          $values = $values[1].split(".");
-          if($values.length === 2)
-          {
-            // $values = something2.something3
-            $value = $values[1];
-            if($value.length >= 2)
-            {
-              $validateType = true;
-            }
-          }
-        }
-      }
-      break;
-    }
-  }
-  return $validateType;
-}
 
-function hideAllErrorAndSuccessMessage()
+function sendDataOnClick(post, url)
 {
-  errorMessage.style.display = 'none';
-  successMessage.style.display = 'none';
-  for(floatingForm of floatingForms)
+  // submitButton.addEventListener("click", event => {
+  //     // contactForm.submit();
+  //     sendData(post, url);
+  //   });
+  submitButton.addEventListener("click", event => {
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/soutenance14/ST/pullRequest?prid=8161867)
```suggestion
  submitButton.addEventListener("click", (event) => {
```